solidity/contracts/BancorNetwork.sol

```diff
@@ -597,9 +617,44 @@ contract BancorNetwork is IBancorNetwork, TokenHolder, ContractRegistryClient {
         }
     }
 
+    // legacy - returns the address of an EtherToken used by the converter
+    function getConverterEtherTokenAddress(IBancorConverter _converter) private view returns (address) {
+        uint256 reserveCount = _converter.connectorTokenCount();
+        for (uint256 i = 0; i < reserveCount; i++) {
+            address reserveTokenAddress = _converter.connectorTokens(i);
+            if (etherTokens[reserveTokenAddress])
+                return reserveTokenAddress;
+        }
+
+        return ETH_RESERVE_ADDRESS;
+    }
+
+    bytes4 private constant GET_RETURN_FUNC_SELECTOR = bytes4(uint256(keccak256("getReturn(address,address,uint256)") >> (256 - 4 * 8)));
+
+    function getReturn(address _dest, address _sourceToken, address _targetToken, uint256 _amount) internal view returns (uint256, uint256) {
+        uint256[2] memory ret;
+        bytes memory data = abi.encodeWithSelector(GET_RETURN_FUNC_SELECTOR, _sourceToken, _targetToken, _amount);
+
+        assembly {
+            let success := staticcall(
+                gas,           // gas remaining
+                _dest,         // destination address
+                add(data, 32), // input buffer (starts after the first 32 bytes in the `data` array)
+                mload(data),   // input length (loaded from the first 32 bytes in the `data` array)
+                ret,           // output buffer
+                64             // output length
+            )
+            if iszero(success) {
+                revert(0, 0)
+            }
+        }
+
+        return (ret[0], ret[1]);
+    }
+
     bytes4 private constant IS_V28_OR_HIGHER_FUNC_SELECTOR = bytes4(uint256(keccak256("isV28OrHigher()") >> (256 - 4 * 8)));

```

Comment:
@yudilevi:

Please fix every occurrence of this:
```
bytes4(uint256(keccak256("...") >> (256 - 4 * 8)))
```
To this:
```
bytes4(keccak256("..."))
```
It will reduce the number of conflicts when you merge it into the base, because I applied that fix just a few hours ago.

Reply:
done